pyomo/core/expr/visitor.py

```diff
@@ -1466,53 +1502,68 @@ def _expression_is_fixed(node):
 #  expression_to_string
 # =====================================================
 
+LEFT_TO_RIGHT = common.OperatorAssociativity.LEFT_TO_RIGHT
+RIGHT_TO_LEFT = common.OperatorAssociativity.RIGHT_TO_LEFT
+
 class _ToStringVisitor(ExpressionValueVisitor):
 
-    def __init__(self, verbose, smap, compute_values):
+    _expression_handlers = None
+
+    def __init__(self, verbose, smap):
         super(_ToStringVisitor, self).__init__()
         self.verbose = verbose
         self.smap = smap
-        self.compute_values = compute_values
 
     def visit(self, node, values):
         """ Visit nodes that have been expanded """
-        tmp = []
+        if node.PRECEDENCE is None:
+            if self._expression_handlers \
+               and node.__class__ in self._expression_handlers:
+                return self._expression_handlers[node.__class__](
+                    self, node, values)
+            return node._to_string(values, self.verbose, self.smap)
+
         for i,val in enumerate(values):
             arg = node._args_[i]
 
             if arg is None:
-                tmp.append('Undefined')                 # TODO: coverage
+                values[i] = 'Undefined'  # TODO: coverage
```

Comment:
Is the TODO still needed? Codecov didn't flag this line...

Reply:
I guess not...